### ISSUE TITLE
custom loss -- weird loss values on validation

### ISSUE THREAD

**Issue by reyhanhgh** [rstudio/keras#1061]
Hi! I am training an LSTM model. I am using the following data generator proposed in the time series tutorial by F. Chollet and J. J. allaire :
```
lookback <- 21
step <- 3
batch_size <- 128
delay <- 1

generator <- function(data, lookback, delay, min_index, max_index, shuffle = FALSE, batch_size, step) {
  if (is.null(max_index))
    max_index <- nrow(data) - delay - 1
  i <- min_index + lookback
  function() {
    if (shuffle) {
      rows <- sample(c((min_index+lookback):max_index), size = batch_size)
    } else {
      if (i + batch_size >= max_index)
        i <<- min_index + lookback
      rows <- c(i:min(i+batch_size-1, max_index))
      i <<- i + length(rows)
    }
    
    samples <- array(0, dim = c(length(rows), lookback / step, dim(data)[[-1]]))
    targets <- array(0, dim = c(length(rows)))
    
    for (j in 1:length(rows)) {
      indices <- seq(rows[[j]] - lookback, rows[[j]]-1, length.out = dim(samples)[[2]])
      samples[j,,] <- data[indices,] 
      targets[[j]] <- data[rows[[j]] + delay, 4]
      }  
    
    list(samples, targets)
  }
}
```
I am trying to create my own `loss function` that has the following formula :

    L = Sum((y_pred - y_true)^2)/Sum((y_true - mean(y_true))^2)

So as its `backend` version, I used :

    fc.loss <- function(y_true, y_pred) {
      dif1 <- y_pred - y_true
      squ1 <- k_pow(dif1, 2)
      sum1 <- k_sum(squ1)
      
      dif2  <- y_true - k_mean(y_true)
      squ2  <- k_pow(dif2, 2)
      sum2  <- k_sum(squ2)
      
      L <- sum1/sum2
      
      return(L)
    }

```
model <- keras_model_sequential() %>% 
  layer_lstm(units = 16, input_shape = list(NULL, dim(data)[[-1]])) %>%
  layer_dense(units = 1)

# ---- train ----
model %>% keras::compile(optimizer= optimizer_adagrad(), loss = fc.loss, metrics = list("mae"))
```
loss values on training are reasonable, well as I expected : they start from 0.95 and decrease gradually over the course of epochs. but it yields weird results on validation : the order of magnitude of `10^8`. Here is the loss-evolution graph :

[![enter image description here][1]][1]

and here the exact values of losses for the first and last f

    Epoch 1/100
    50/50 [==============================] - 7s 136ms/step - loss: 0.9513 - mae: 0.3990 - val_loss: 1023264317186.3749 - val_mae: 0.2399
    Epoch 2/100
    50/50 [==============================] - 7s 136ms/step - loss: 0.8550 - mae: 0.3430 - val_loss: 837731505269.4755 - val_mae: 0.2188
    Epoch 3/100
    50/50 [==============================] - 7s 138ms/step - loss: 0.7501 - mae: 0.2950 - val_loss: 741010940624.6019 - val_mae: 0.2051
    Epoch 4/100
    50/50 [==============================] - 7s 136ms/step - loss: 0.7339 - mae: 0.2887 - val_loss: 703123064973.9722 - val_mae: 0.2027
    Epoch 5/100
    50/50 [==============================] - 7s 136ms/step - loss: 0.6935 - mae: 0.2618 - val_loss: 476261721396.7943 - val_mae: 0.2036

    Epoch 95/100
    50/50 [==============================] - 6s 128ms/step - loss: 0.5273 - mae: 0.1777 - val_loss: 17348635228.9776 - val_mae: 0.1790
    Epoch 96/100
    50/50 [==============================] - 6s 126ms/step - loss: 0.5429 - mae: 0.1710 - val_loss: 16628514948.4826 - val_mae: 0.1730
    Epoch 97/100
    50/50 [==============================] - 6s 128ms/step - loss: 0.5525 - mae: 0.1849 - val_loss: 18021322145.4323 - val_mae: 0.2002
    Epoch 98/100
    50/50 [==============================] - 7s 130ms/step - loss: 0.5930 - mae: 0.1969 - val_loss: 17155417718.3754 - val_mae: 0.1937
    Epoch 99/100
    50/50 [==============================] - 6s 128ms/step - loss: 0.5891 - mae: 0.2038 - val_loss: 17123782552.8853 - val_mae: 0.1849
    Epoch 100/100
    50/50 [==============================] - 6s 128ms/step - loss: 0.5782 - mae: 0.1741 - val_loss: 16684981572.5142 - val_mae: 0.1780

I don't understand why it behaves like this. Where is my mistake ? 
@dfalbel @skeydan @jjallaire do you have any thought on this please ? I would highly appreciate your help. 

  [1]: https://i.stack.imgur.com/wT41f.png
